### PR TITLE
Fix/68966 - [ORTF] report:enddate

### PR DESCRIPTION
**Pull request by mohammadjafarinejad**
### Explanation of Change

Add support for {report:enddate} – Report end date (newest expense date)

### Fixed Issues

$ https://github.com/Expensify/App/issues/68966
PROPOSAL: https://github.com/Expensify/App/issues/68966#issuecomment-3212172829

### Tests

Same as QA.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Turn on `Force offline mode` and do the same steps as in the QA Steps section.
Report titles should be properly formatted.

### QA Steps

Before running tests, ensure your account has access to BETA functionality `useCustomReportNamesNewExpensify`. To add yourself to the beta, please follow the pre-testing steps as described in [this document](https://docs.google.com/document/d/1q6T1rOSgGu6H29a5TdnW5obvvkAiGE5QeDtlayWNxw8/edit?usp=sharing).

#### Note:
There is a new rNVP logic: [Expensify/App#67823](https://github.com/Expensify/App/issues/67823).  
For now, make sure to **create the report while online** so the rNVP value is stored — this ensures the formula calculation works correctly.

In some cases, the **report name may show stale data**. This issue is not related to this PR and will be fixed in a future update when we start using **Onyx Derived Values** for report name calculation.

---

### **Precondition**

1. Navigate to **Workspaces**.
2. Select a **workspace** where you have **admin rights**.
3. Go to the **Reports** section.
4. Edit the **default report title formula** to:

   ```
   Started {report:startdate} vs Ended {report:enddate}
   ```
5. **Save** the formula.

---

#### **Test Case 1 – Date Fields Comparison**

1. Create a **new report** via FAB.
2. Add **two expense transactions** with dates **different from the report creation date**.
3. Verify that the **report title** shows both **start date** and **end date** correctly.

---

#### **Test Case 2 – Transaction Addition Impact**

1. Create a **new report** via FAB.
2. Add the **first expense transaction** and note the **report title**.
3. Add a **second expense transaction** with a **different date**.
4. Verify that the **report title** updates to reflect the correct **start date** and **end date**.

---

#### **Test Case 3 – Transactions**

1. Create a **new report** and add **two transactions**:
   a. First transaction with date `2025-08-16`
   b. Second transaction with date `2025-08-17`
2. Verify the **report title** equals:

   ```
   Started 2025-08-16 Ended 2025-08-17
   ```
3. Go to **Account → Troubleshoot** and enable **Force Offline** option.
4. Return to the newly created report.
5. Edit the **second transaction's date** to `2025-08-14`.
6. Verify the **report title** updates to:

   ```
   Started 2025-08-14 Ended 2025-08-16
   ```
7. Change the **second transaction's date** back to `2025-08-17`.
8. Verify the **report title** updates back to:

   ```
   Started 2025-08-16 Ended 2025-08-17
   ```
9. Add a **new transaction** with date `2025-08-05`.
10. Verify the **report title** updates to:

    ```
    Started 2025-08-05 Ended 2025-08-17
    ```

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified there are no new alerts related to the `canBeMissing` param for `useOnyx`
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L80)
      - [x] If any non-english text was added/modified, I used [JaimeGPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) to get English > Spanish translation. I then posted it in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L116-L123)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If new assets were added or existing ones were modified, I verified that:
    - [x] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [x] The assets load correctly across all supported platforms.
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/a3e01865-493e-4fb7-a25c-3a87cbaf9167

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/da867dfe-92fa-4e67-8900-fd7559337b6e

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/bd8afc1f-aecf-45d5-ab56-040dd9e0d4aa

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/f8232961-f85c-4361-976f-47d2472bab74

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/8dd66e2b-8a26-4528-a248-25e5500c6cc6

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/5587bdaf-1233-4b2d-9309-41205279abd6

</details>

### CODE REVIEW

File: src/libs/OptimisticReportNames.ts [hunk 1]
```diff
@@ -167,6 +167,20 @@ function isValidReportType(reportType?: string): boolean {
     );
 }
 
+function getNewReportFromUpdates(reportID: string, updates: OnyxUpdate[]): Report | undefined {
```
Comment (mohammadjafarinejad) [inline src/libs/OptimisticReportNames.ts]: @arosiclair @linhvovan29546
This can fix the issue. Please let me know what do you think

https://github.com/Expensify/App/issues/68966#issuecomment-3301975835

File: src/libs/Formula.ts [hunk 2]
```diff
@@ -473,25 +491,22 @@ function getOldestTransactionDate(reportID: string, context?: FormulaContext): s
...
     }
 
     let oldestDate: string | undefined;
 
     transactions.forEach((transaction) => {
-        // Use updated transaction data if available and matches this transaction
-        const currentTransaction = context?.transaction && transaction.transactionID === context.transaction.transactionID ? context.transaction : transaction;
-
-        const created = getCreated(currentTransaction);
+        const created = getCreated(transaction);
```
Comment (linhvovan29546) [inline src/libs/Formula.ts:509]: Can you clarify the purpose of updating this line?

File: src/libs/Formula.ts [hunk 3]
```diff
@@ -465,6 +467,22 @@ function formatType(type: string | undefined): string {
     return typeMapping[type.toLowerCase()] || type;
 }
 
+function getAllReportTransactionsWithContext(reportID: string, context?: FormulaContext): Transaction[] {
```
Comment (linhvovan29546) [inline src/libs/Formula.ts:477]: Can you clarify the purpose of this function?
Also, please add a comment to describe this function.

File: src/libs/OptimisticReportNames.ts [hunk 4]
```diff
@@ -167,6 +167,20 @@ function isValidReportType(reportType?: string): boolean {
...
+    if (!reportID) {
+        return undefined;
+    }
+    const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${reportID}`;
+    const newReportUpdate = updates.find((u) => u.key === reportKey && u.onyxMethod === Onyx.METHOD.SET);
```
Comment (linhvovan29546) [inline src/libs/OptimisticReportNames.ts]: ` const newReportUpdate = updates.find((update) => update.key === reportKey && update.onyxMethod === Onyx.METHOD.SET);`

**Review by linhvovan29546 — CHANGES_REQUESTED** [review]

File: src/libs/Formula.ts (hunk 3 above)
Comment (mohammadjafarinejad) [reply src/libs/Formula.ts:477]: `getAllReportTransactionsWithContext()` gets all Onyx transactions + adds/updates any transaction from context. This fixes the issue you mentioned [here](https://github.com/Expensify/App/issues/68966#issuecomment-3283685919).

Sure, I'll add a comment.

File: src/libs/Formula.ts (hunk 2 above)
Comment (mohammadjafarinejad) [reply src/libs/Formula.ts:509]: Because we are using `getAllReportTransactionsWithContext()`, we don't need to update it anymore.

File: src/libs/OptimisticReportNames.ts (hunk 4 above)
Comment (mohammadjafarinejad) [reply src/libs/OptimisticReportNames.ts]: Thanks!

File: src/libs/OptimisticReportNames.ts [hunk 5]
```diff
@@ -167,6 +167,20 @@ function isValidReportType(reportType?: string): boolean {
     );
 }
 
+function getNewReportFromUpdates(reportID: string, updates: OnyxUpdate[]): Report | undefined {
+    if (!reportID) {
+        return undefined;
+    }
+    const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${reportID}`;
+    const newReportUpdate = updates.find((update) => update.key === reportKey && update.onyxMethod === Onyx.METHOD.SET);
```
Comment (linhvovan29546) [inline src/libs/OptimisticReportNames.ts]: I don't think` checking update.onyxMethod === Onyx.METHOD.SET` is a good idea, because we can't reliably track the value of `onyxMethod` elsewhere. This could lead to regressions in the future(If we use Onyx.Method.MERGE).

**Review by linhvovan29546 — CHANGES_REQUESTED** [review]

File: src/libs/OptimisticReportNames.ts (hunk 5 above)
Comment (mohammadjafarinejad) [reply src/libs/OptimisticReportNames.ts]: I completely agree, but we also need an alternative way to ensure we are getting the most complete report from the `updates` (and not a partial one), or a related one to the transaction (newly created one).

In `computeReportNameIfNeeded()`, it is currently using the update as a `Report` for the `report` case, but in the `transaction` case, the problem is that we are searching through the entire update, and some of these may be partial reports. We also shouldn't forget about the updates ordering here.

https://github.com/Expensify/App/blob/4df2bd186901cad43acad2642a07ead7c69fab6c/src/libs/OptimisticReportNames.ts#L187-L191

I will find a better way of handling this and will let you know. In the meantime, I would appreciate it if you could provide any suggestions for a better approach.

Example of another way that we can handle:
```tsx
    const newReport = newReportUpdate.value as Partial<Report>;
    if (newReport.pendingFields?.createChat !== CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD) {
        return undefined;
    }

    return newReport as Report;
```

_Also, these are the logs for the `OnyxUpdates` in `updateOptimisticReportNamesFromUpdates()` when a new transaction and a new report appear in the same batch of updates._

[logs.json](https://github.com/user-attachments/files/22429376/logs.json)

Comment (linhvovan29546) [reply src/libs/OptimisticReportNames.ts]: If this is used to fix an issue in `offline` mode, then I think using `pendingField` is appropriate. However, I also suggest adding a comment where `pendingField` is added, so it can be tracked more easily

Comment (mohammadjafarinejad) [reply src/libs/OptimisticReportNames.ts]: When we’re online, this issue doesn’t appear because other updates call `compute()` right away. But offline, nothing triggers `compute()` right away, so the problem becomes visible. I think it can also happen on slow networks.

The most reliable fix is to validate the update. We have two options:

1. Check for `pendingFields` and add a comment where we create a new optimistic report along with a new transaction in `requestMoney()`.

2. Use the new `isPartialReport()` to ensure the update includes all essential fields.

I think if we can validate that an update has the properties of an `OptimisticReport`, we could reliably treat it as either a full `Report` or an `optimistic` version.

Also, currently, if `computeReportNameIfNeeded()` cannot find the report in Onyx, it treats the update as a `Report` without validation, even though it might actually be an `OptimisticExpenseReport` or partial report. Later, it only checks `isValidReportType()`. I think since `report.type` doesn’t change much, if a report update includes `type`, the chance of it being partial is rare.

https://github.com/Expensify/App/blob/4df2bd186901cad43acad2642a07ead7c69fab6c/src/libs/OptimisticReportNames.ts#L190-L191

So if an `update` passes all these checks, we can safely treat it as a `Report` that meets the computation requirements.

Please tell me what you think and which option you prefer we go with.

```tsx
/**
 * Finds a complete new report from updates batch.
 */
function getNewReportFromUpdates(reportID: string, updates: OnyxUpdate[]): Report | undefined {
    if (!reportID) {
        return undefined;
    }
    const reportKey = `${ONYXKEYS.COLLECTION.REPORT}${reportID}`;

    const newReport = updates.find((update) => {
        if (update.key !== reportKey) {
            return false;
        }

        const report = update.value as Partial<Report> | undefined;
        if (!report || !isValidReportType(report.type)) {
            return false;
        }

        // Option 1: Check for pending creation fields (optimistic update)
        const hasCreateChatPending = report.pendingFields?.createChat === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD;
        const hasCreateReportPending = report.pendingFields?.createReport === CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD;
        if (!hasCreateChatPending && !hasCreateReportPending) {
            return false;
        }

        // Option 2: Check if report has all essential fields
        if (isPartialReport(report)) {
            return false;
        }

        return true;
    })?.value as Report | undefined;

    return newReport;
}

/**
 * Checks if a report is partial/incomplete by validating essential fields.
 * Based on OptimisticNewReport and OptimisticExpenseReport type definitions.
 */
function isPartialReport(report: Partial<Report | OptimisticChatReport>): boolean {
    if (!report) {
        return true;
    }

    // Check for essential fields that should be present in a complete report
    const hasEssentialFields =
        report.reportID &&
        report.type &&
        report.ownerAccountID !== undefined &&
        report.stateNum !== undefined &&
        report.statusNum !== undefined;

    return !hasEssentialFields;
}
```

Comment (linhvovan29546) [reply src/libs/OptimisticReportNames.ts]: I think using `isPartialReport` makes sense, but I'm not sure if the condition in hasEssentialFields is correct. Are you certain the condition is accurate?

Comment (mohammadjafarinejad) [reply src/libs/OptimisticReportNames.ts]: Yes, in all functions where we build a new optimistic report, such as `buildOptimisticIOUReport()` and `buildOptimisticExpenseReport()`, we are setting all of those fields. I also added a better comment explaining this.

Comment (linhvovan29546) [reply src/libs/OptimisticReportNames.ts]: This doesn't work if I try to delete a report while offline.

Comment (mohammadjafarinejad) [reply src/libs/OptimisticReportNames.ts]: Could you please clarify the issue or share the specific steps to reproduce it? Deleting a report works fine. Are you referring to deleting an expense, not updating the report name?

Comment (linhvovan29546) [reply src/libs/OptimisticReportNames.ts]: Ah, I forgot to upload the video

Comment (linhvovan29546) [reply src/libs/OptimisticReportNames.ts]: https://github.com/user-attachments/assets/5e910e97-fa43-451a-a868-a61a40d41c05

Comment (mohammadjafarinejad) [reply src/libs/OptimisticReportNames.ts]: Thank you for the video! I see the issue now. You're right - we need to filter out transactions with `pendingAction` delete in both `getOldestTransactionDate()` and `getNewestTransactionDate()` functions. I'll fix this and update the PR.

File: src/libs/Formula.ts [hunk 6]
```diff
@@ -473,25 +495,25 @@ function getOldestTransactionDate(reportID: string, context?: FormulaContext): s
...
     transactions.forEach((transaction) => {
-        // Use updated transaction data if available and matches this transaction
-        const currentTransaction = context?.transaction && transaction.transactionID === context.transaction.transactionID ? context.transaction : transaction;
-
-        const created = getCreated(currentTransaction);
+        const created = getCreated(transaction);
         if (!created) {
             return;
         }
+        if (transaction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
```
Comment (linhvovan29546) [inline src/libs/Formula.ts:514]: NAB: Could you add comments to this line for better clarity?

File: src/libs/OptimisticReportNames.ts (hunk 5 above)
Comment (linhvovan29546) [reply src/libs/OptimisticReportNames.ts]: I am still able to reproduce this bug while offline.

https://github.com/user-attachments/assets/17cda97d-d165-4f78-8ee2-d27a8cef137e

These conditions are not met
Screenshot 2025-09-23 at 17 13 59

**Review by linhvovan29546 — CHANGES_REQUESTED** [review]

File: src/libs/Formula.ts (hunk 6 above)
Comment (mohammadjafarinejad) [reply src/libs/Formula.ts:514]: Sure

Comment (mohammadjafarinejad) [reply src/libs/Formula.ts:514]: We can also refactor it like this to avoid code duplication:

```tsx
function getValidTransactionsWithDates(reportID: string, context?: FormulaContext): string[] {
    if (!reportID) {
        return [];
    }

    const transactions = [...getReportTransactions(reportID)];
    const contextTransaction = context?.transaction;

    if (contextTransaction?.transactionID && contextTransaction.reportID === reportID) {
        const transactionIndex = transactions.findIndex(
            (transaction) => transaction?.transactionID === contextTransaction.transactionID
        );
        
        if (transactionIndex >= 0) {
            transactions[transactionIndex] = contextTransaction;
        } else {
            transactions.push(contextTransaction);
        }
    }  

    const validDates: string[] = [];
    
    transactions.forEach((transaction) => {
        const created = getCreated(transaction);
        if (!created) {
            return;
        }
        if (transaction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
            return;
        }
        if (isPartialTransaction(transaction)) {
            return;
        }
        
        validDates.push(created);
    });

    return validDates;
}

function getOldestTransactionDate(reportID: string, context?: FormulaContext): string | undefined {
    const validDates = getValidTransactionsWithDates(reportID, context);
    
    if (validDates.length === 0) {
        return new Date().toISOString();
    }
    
    return validDates.reduce((oldest, current) => current < oldest ? current : oldest);
}
```

Comment (linhvovan29546) [reply src/libs/Formula.ts:514]: @mohammadjafarinejad Please check my comment here as well https://github.com/Expensify/App/pull/70627#discussion_r2371824853

File: src/libs/Formula.ts [hunk 7]
```diff
@@ -476,25 +498,26 @@ function getOldestTransactionDate(reportID: string, context?: FormulaContext): s
...
-        // Use updated transaction data if available and matches this transaction
-        const currentTransaction = context?.transaction && transaction.transactionID === context.transaction.transactionID ? context.transaction : transaction;
-
-        const created = getCreated(currentTransaction);
+        const created = getCreated(transaction);
         if (!created) {
             return;
         }
+        // Skip transactions with pending deletion (offline deletes) to calculate dates properly.
+        if (transaction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
```
Comment (linhvovan29546) [inline src/libs/Formula.ts:514]: @mohammadjafarinejad  Is these change necessary? I was under the impression that we would apply the change similar to getOldestTransactionDate, meaning we only need [getNewestTransactionDate](https://github.com/Expensify/App/pull/70627/files#diff-e14b78c040cd62ae05ad32b9a6b144534b30661796a5e021babf9ec5e905358aR251-R252)

Comment (mohammadjafarinejad) [reply src/libs/Formula.ts:514]: This change is related to https://github.com/Expensify/App/pull/70627#discussion_r2368044843. Otherwise, when a user deletes a transaction, the title doesn’t display the correct date.

Comment (linhvovan29546) [reply src/libs/Formula.ts:514]: This also occurs on the main branch with `report:startdate`, so I think we shouldn't address it here, as we plan to refactor this in a future

Comment (linhvovan29546) [reply src/libs/Formula.ts:514]: I mean, since we plan to redesign the report name, I think we should focus only on `report:enddate` for simplicity

Comment (mohammadjafarinejad) [reply src/libs/Formula.ts:514]: Please check [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1760506589203779?thread_ts=1760387161.873509&cid=C01GTK53T8Q). The internal team wants to revert the rNVP check so they can test it manually under beta. Also, I think not all edge cases were covered in the original implementation.

So, do you think we should go ahead and remove it?

Comment (linhvovan29546) [reply src/libs/Formula.ts:514]: @arosiclair Since we're [redesigning](https://expensify.slack.com/archives/C01GTK53T8Q/p1760489662769689?thread_ts=1760387161.873509&cid=C01GTK53T8Q) the report name, I think we should focus on implementing `report:enddate` and the related unit tests in this PR. We can ignore unrelated changes(offline mode) for now . What do you think ?

Comment (arosiclair) [reply src/libs/Formula.ts:514]: These optimistic functions are intended to be used while offline so I think we should keep this even if we're making some refactors elsewhere.

**Comment by linhvovan29546** [comment]
@mohammadjafarinejad You've missed adding a screenshot/recording to the PR description

**Review by linhvovan29546 — APPROVED** [review]

**Comment by linhvovan29546** [comment]
> @linhvovan29546 Should we add a note in QA steps about rNVP?

 👍

**Review by arosiclair — APPROVED** [review]
